### DISCUSSION TITLE
Getting Nuxt to play nicely with mswjs

### DISCUSSION THREAD

**Discussion by amrnn90** [nuxt/nuxt#24519] [Questions]
Hello, I am not sure if I should make this an issue or not, so posting here for now.

I am trying to integrate [mswjs](https://mswjs.io/) with Nuxt. Here is a minimal example of what I got so far:
https://stackblitz.com/edit/nuxt-starter-wd58t8?file=app.vue

I am facing two issues:

**1. Nuxt's default `$fetch` does not seem to work with `mswjs`, so I had to use `createFetch()` in my plugin**
```ts
// ~/plugins/msw.ts

import { createFetch } from 'ofetch';

export default defineNuxtPlugin(async (nuxtApp) => {
  if (process.server) {
    const server = (await import('~/mocks/node')).server;
    server.listen();
    console.log('msw server listening...');
  }
  if (process.client) {
    const { worker } = await import('~/mocks/browser');
    await worker.start();
    console.log('msw worker started...');
  }
  // @ts-ignore
  // had to add this since Nuxt's $fetch is not working with msw
  globalThis.$myfetch ||= createFetch();
});
```

Which I later use in pages:
```vue
// App.vue

<script setup lang="ts">
const { data: posts } = useFetch('http://localhost:3000/posts', {
  $fetch: $myfetch,
});
// This does not work
// const { data: posts } = useFetch('http://localhost:3000/posts');
</script>

<template>
  <div>
    <h1>Posts</h1>
    <pre>{{ posts }}</pre>
  </div>
</template>
```
**2. Updating the mock data in `~/mocks/handlers.ts` does not seem to be taking effect, even after refreshing the page. I have to restart the dev server manually for it to take effect, not sure if this is an issue with Nuxt or mswjs or with my code.**

Any help is appreciated, thanks!

**Comment by stichingsd-vitrion** [comment]
Would be lovely is this can be picked up

Currently we are stuck bcs of https://github.com/unjs/ofetch/issues/295
